### ISSUE TITLE
GRPO online example not reproducible

### ISSUE THREAD

**Issue by JakobJBauer** [huggingface/trl#3406]
### Reproduction

According to the guide on [https://huggingface.co/docs/trl/main/en/grpo_trainer](https://huggingface.co/docs/trl/main/en/grpo_trainer), we achieve convergence after ~2.5k global steps, and the converging process seems very smooth. Also it is reported that training takes 24h on 8 GPUS.

When running the exact same code (without accelerate) on a single A100 I have:
- much faster runtime
- much quicker convergence
- but the model is not nearly as accurate
- and the convergence process is not nearly as smooth

The code I copied and directly ran on A100 is:
```python
from datasets import load_dataset
from trl import GRPOConfig, GRPOTrainer

dataset = load_dataset("trl-lib/tldr", split="train")

# Define the reward function, which rewards completions that are close to 20 characters
def reward_len(completions, **kwargs):
    return [-abs(20 - len(completion)) for completion in completions]

training_args = GRPOConfig(output_dir="Qwen2-0.5B-GRPO", logging_steps=10)
trainer = GRPOTrainer(
    model="Qwen/Qwen2-0.5B-Instruct",
    reward_funcs=reward_len,
    args=training_args,
    train_dataset=dataset,
)
trainer.train()
```

And according to the guide, the plots look like this:
![Image](https://github.com/user-attachments/assets/49cfa780-6960-49d6-b6be-0d033d11a9f0)

Now there are two main issues:
1. I don't understand the plots. Why does the reward go from 0 to 1, when in the code itself, the reward is always negative or zero?
2. My plots look way different. Here is something that is comparable:
![Image](https://github.com/user-attachments/assets/36af701a-a61e-47ab-ac69-fc48b6dc7049)
![Image](https://github.com/user-attachments/assets/309ce053-a54d-4d72-b05c-ec61bcac91b8)

Has this to do with distributed training? Why does the code converge so much faster (but not so beautifully) on my machine? Peak reward on my test run is -0.15 after only 290 steps.

### System Info

- Platform: Linux-5.15.0-134-generic-x86_64-with-glibc2.35
- Python version: 3.11.6
- TRL version: 0.16.1
- PyTorch version: 2.6.0
- CUDA device(s): not available
- Transformers version: 4.51.3
- Accelerate version: 1.6.0
- Accelerate config: not found
- Datasets version: 3.5.0
- HF Hub version: 0.30.2
- bitsandbytes version: not installed
- DeepSpeed version: not installed
- Diffusers version: not installed
- Liger-Kernel version: not installed
- LLM-Blender version: not installed
- OpenAI version: 1.75.0
- PEFT version: not installed
- vLLM version: 0.8.4

### Checklist

- [x] I have checked that my issue isn't already filed (see [open issues](https://github.com/huggingface/trl/issues?q=is%3Aissue))
- [x] I have included my system information
- [x] Any code provided is minimal, complete, and reproducible ([more on MREs](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any code provided is properly formatted in code blocks, (no screenshot, [more on code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any traceback provided is complete

**Comment by shirinyamani** [comment]
Thanks for trying! @JakobJBauer 
As @LeonEricsson correctly pointed, the figures you see on the documentation is not the output of the code snippet. A simple way to figure that is for such bit total step that only took ~1 day, we can say that `use_vllm=True` in that example! However, the  training_args of the code snippet where we define the GRPOConfig does not even have 'use_vllm=True'. Thanks for bringing this to our attention and btw we provided a guide on grpo generation using vllm if you set the  `vllm_mode="server"` [here](https://huggingface.co/docs/trl/main/en/vllm_integration). The example used in this link has detail of the setup including the parallelism setup. Hope this example helps you.  

training_args = GRPOConfig(
    ...,
    use_vllm=True,
    vllm_mode="server", 
)
